bblocks/import_tools/world_bank.py

```diff
@@ -231,98 +215,151 @@ def _clean_pink_sheet(df: pd.DataFrame, sheet: str) -> pd.DataFrame:
             "Precious Metals",
         ]
         df["period"] = pd.to_datetime(df.period, format="%YM%m")
+    df = pd.melt(df, id_vars="period", var_name="indicator", value_name="value")
 
-        return df
-
-    else:
-        raise ValueError(
-            "invalid sheet name. "
-            "Please specify 'Monthly Indices' or 'Monthly Prices'"
-        )
+    return df
 
 
 @dataclass
-class WorldBankPinkSheet:
-    """An object to help download data from World Bank Pink sheets
+class PinkSheet(ImportData):
+    """An object to help download data from World Bank Pink sheets.
+
+    In order to use, create an instance of this class, specifying the sheet name - 'Monthly Prices', 'Monthly Indices'.
+    Then, call the load_indicator method to load an indicator, optionally specifying in indicator or
+    list of indicators. This can be done multiple times. If the data has never been downloaded,
+    it will be downloaded. If it has been downloaded, it will be loaded from disk.
+    If `update_data` is set to True when creating the object, the full dataset will be downloaded to disk
+    when `load_indicator` is called for the first time.
+    You can force an update by calling `update` if you want to refresh the data stored on disk.
+    You can get a dataframe of the data by calling `get_data`
 
-    Parameters:
-        sheet (str): name of the sheet in the Pink Sheet ['Monthly Prices', 'Monthly Indices']
     """
 
-    sheet: str
-    data: pd.DataFrame = None
-    update_data: bool = False
+    sheet: str = None
 
-    def __post_init__(self) -> None:
-        if self.update_data:
-            self.update()
-        if not os.path.exists(f"{PATHS.imported_data}/{self.file_name}"):
+    def __post_init__(self):
+        if self.sheet not in ["Monthly Indices", "Monthly Prices"]:
+            raise ValueError(
+                "Invalid sheet name. "
+                "Please specify 'Monthly Indices' or 'Monthly Prices'"
+            )
+        self.indicators = {}
```

Comment:
The parent class defines the indicators variable. There is no need to reinstantiate it here.